File: ALDS1/ALDS1_1_C.c
#include <stdio.h>
#include <math.h>

int main(void){
    int num = 0;
    printf("2以上の自然数を入力\n");
    scanf("%d", &num);

    if (num == 2){
        printf("素数です¥n");
    }else if ((num < 2) || ( (num % 2) == 0)){
        printf("素数ではないです\n");
    }else{
        int i = 3;
        do{
            if ( (num % i) == 0 ){
                printf("素数ではないです\n");
            }
            i += 2;

        }while ( i <= sqrt(num));
        printf("素数です\n");

    }

    return 0;
}

File: ALDS1/ALDS1_1_C.py
#coding:utf-8
import math

def prime(n):
    if n == 2:
        return True
    elif (n < 2) or (n % 2 == 0):
        return  False
    else:
        i = 3
        while i <= math.sqrt(n):
            if n % i == 0:
                return False
            i += 2

        return True


if __name__ == "__main__":
    x = int(input())
    counter = 0

    for i in range(x):
        n = int(input())
        if prime(n) == True:
            counter += 1

    print(counter)
